ams/optimizer/common/src/main/java/com/netease/arctic/optimizer/common/OptimizerToucher.java

```diff
@@ -70,6 +74,11 @@ private boolean checkToken() {
         LOG.info("Registered optimizer to ams with token:{}", token);
         return true;
       } catch (TException e) {
+        if (e instanceof ArcticException
+            && ErrorCodes.FORBIDDEN_ERROR_CODE == ((ArcticException) e).getErrorCode()) {
+          stop(); // Don't need to getToken again
```

Comment:
Can I use `System.exit(1)` instead `stop()`?  Since if we throw the exception out, there are many code needed to be modified.

Reply:
I'm not sure if it's appropriate for operators to exist directly in Flink/Spark, and what kind of results it might lead to.

Reply:
Maybe we can wrap the exception to RuntimeException to avoid changing existing methods.

Reply:
> I'm not sure if it's appropriate for operators to exist directly in Flink/Spark, and what kind of results it might lead to.

It's OK. For the flink/spark-based optimizer, we give right error logging info and right exit code which is used to judge the running result for the schedule platform. 0 stands for succeed, other code stands for failed